File: frontend/src/PrescriptionPDF.js
import React from 'react';
import jsPDF from 'jspdf';

function PrescriptionPDF({ data }) {
  const generatePDF = () => {
    const doc = new jsPDF();
    doc.setFontSize(12);
    doc.text('Ordonnance Chimiothérapie', 20, 20);

    const safe = (val) => (val !== null && val !== undefined && val !== '' ? val : '-');

    doc.text(`Patient : ${safe(data.patient)}`, 20, 30);
    doc.text(`Stade : ${safe(data.stade)}`, 20, 40);
    doc.text(`Diagnostic : ${safe(data.diagnosis)}`, 20, 50);
    doc.text(`Protocole : ${safe(data.protocol)}`, 20, 60);
    doc.text(`Poids : ${safe(data.weight_kg)} kg`, 20, 70);
    doc.text(`Taille : ${safe(data.height_cm)} cm`, 20, 80);
    doc.text(`BSA : ${safe(data.bsa)} m²`, 20, 90);

    let y = 100;
    doc.text('Posologie :', 20, y);
    y += 10;

    if (data.dosage) {
      Object.entries(data.dosage).forEach(([drug, dose]) => {
        doc.text(`• ${drug} : ${dose} mg`, 25, y);
        y += 8;
      });
    } else {
      doc.text('Aucune posologie calculée.', 25, y);
      y += 8;
    }

    y += 10;
    doc.text(`Fait le : ${new Date().toLocaleDateString()}`, 20, y);
    y += 10;
    doc.text(`Médecin : ${safe(data.doctor)}`, 20, y);

    doc.save('ordonnance.pdf');
  };

  return (
    <button onClick={generatePDF} className="send-btn" style={{ marginTop: "10px" }}>
      📄 Télécharger Ordonnance (PDF)
    </button>
  );
}

export default PrescriptionPDF;

File: backend/prescription_pdf.py
from fpdf import FPDF

def generate_pdf(data, output_stream):
    pdf = FPDF()
    pdf.add_page()
    pdf.set_font("Arial", size=12)
    pdf.cell(200, 10, txt="Ordonnance Chimiothérapie", ln=1)

    def safe(val): return str(val) if val else '-'

    pdf.cell(200, 10, txt=f"Patient : {safe(data.get('patient'))}", ln=1)
    pdf.cell(200, 10, txt=f"Cycle : {safe(data.get('cycle'))}", ln=1)
    pdf.cell(200, 10, txt=f"Diagnostic : {safe(data.get('diagnosis'))}", ln=1)
    pdf.cell(200, 10, txt=f"Protocole : {safe(data.get('protocol'))}", ln=1)
    pdf.cell(200, 10, txt=f"Poids : {safe(data.get('weight_kg'))} kg", ln=1)
    pdf.cell(200, 10, txt=f"Taille : {safe(data.get('height_cm'))} cm", ln=1)
    pdf.cell(200, 10, txt=f"BSA : {safe(data.get('bsa'))} m²", ln=1)

    pdf.cell(200, 10, txt="Posologie :", ln=1)
    dosage = data.get('dosage', {})
    for drug, dose in dosage.items():
        pdf.cell(200, 10, txt=f"• {drug} : {dose} mg", ln=1)

    pdf.cell(200, 10, txt=f"Médecin : {safe(data.get('doctor'))}", ln=1)

    pdf.output(output_stream)
